dask/array/overlap.py

```diff
@@ -536,16 +577,30 @@ def coerce_depth(ndim, depth):
         depth = (depth,) * ndim
     if isinstance(depth, tuple):
         depth = dict(zip(range(ndim), depth))
-
+    if isinstance(depth,dict):
+        for i in range(ndim):
+            if i not in depth:
+                depth.update({i:0})
     return depth
 
 
 def coerce_boundary(ndim, boundary):
+    default = 'reflect'
     if boundary is None:
-        boundary = 'reflect'
+        boundary = default
```

Comment:
As `boundary` isn't a keyword argument for `coerce_boundary`, I think the use of a `default` value doesn't apply here and I'd prefer to stick with the original 

```python
if boundary is None:
    boundary = 'reflect'
```

If the concern here is that the `boundary = None` -> `'reflect'` logic is unintuitive, we could update function docstrings appropriately or consider changing the default value usage of `boundary=None` to `boundary='reflect'`.

Reply:
I ended up leaving it as is.  boundary = None -> 'reflect'.  After not hearing back right away, I assumed that changing the default value usage might cause backwards-compatability issues, so my intention was to update the docstring to assert that boundary='none' should be used for all asymmetric overlap use cases.

Reply:
It might make sense to make a breaking change here in 2.0.

OTOH it might make sense to eliminate these arguments and push users to use `da.pad` in conjunction with `da.map_overlap` (using no boundary condition) instead. ( https://github.com/dask/dask/issues/4252 ) That would allow asymmetric overlap to work on all boundary cases.

Reply:
I don't disagree.